tests/syslog/test_syslog_rate_limit.py

```diff
@@ -0,0 +1,239 @@
+import logging
+import os
+import pytest
+import random
+
+from tests.common.config_reload import config_reload
+from tests.common.utilities import skip_release
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.plugins.loganalyzer.loganalyzer import LogAnalyzer
+from tests.common.helpers.sonic_db import SonicDbCli
+
+logger = logging.getLogger(__name__)
+
+RATE_LIMIT_BURST = 100
+RATE_LIMIT_INTERVAL = 10
+# Generate 101 packets in tests/syslog/log_generator.py, so that 1 log message will be dropped by rsyslogd
+LOG_MESSAGE_GENERATE_COUNT = 101
+BASE_DIR = os.path.dirname(os.path.abspath(__file__))
+LOCAL_LOG_GENERATOR_FILE = os.path.join(BASE_DIR, 'log_generator.py')
+REMOTE_LOG_GENERATOR_FILE = os.path.join('/tmp', 'log_generator.py')
+DOCKER_LOG_GENERATOR_FILE = '/log_generator.py'
+# rsyslogd prints this log when rate-limiting reached
+LOG_EXPECT_SYSLOG_RATE_LIMIT_REACHED = '.*begin to drop messages due to rate-limiting.*'
+# Log pattern for tests/syslog/log_generator.py
+LOG_EXPECT_LAST_MESSAGE = '.*{}rate-limit-test: This is a test log:.*'
+
+pytestmark = [
+    pytest.mark.topology("any")
+]
+
+
+@pytest.fixture(autouse=True, scope="module")
+def check_image_version(rand_selected_dut):
+    """Skips this test if the SONiC image installed on DUT is older than 202205
+
+    Args:
+        duthost: Hostname of DUT.
+
+    Returns:
+        None.
+    """
+    skip_release(rand_selected_dut, ["201811", "201911", "202012", "202106", "202205"])
```

Comment:
Can you use the conditional mark plugin to do the skipping?

Reply:
I did not find such conditional mark plugin. Do you mean to create one?